doc/source/basics.rst

```diff
@@ -64,14 +64,28 @@ Note, **these attributes can be safely assigned to**!
    df.columns = [x.lower() for x in df.columns]
    df
 
-To get the actual data inside a data structure, one need only access the
-**values** property:
+Pandas objects (:class:`Index`, :class:`Series`, :class:`DataFrame`) can be
+thought of as containers for arrays, which hold the actual data and do the
+actual computation. For many types, the underlying array is a
+:class:`numpy.ndarray`. However, pandas and 3rd party libraries may *extend*
+NumPy's type system to add support for custom arrays
+(see :ref:`dsintro.data_types`).
+
+To get the actual data inside a :class:`Index` or :class:`Series`, use
+the **array** property
+
+.. ipython:: python
+
+   s.array
+   s.index.array
+
+Getting the "raw data" inside a :class:`DataFrame` is possibly a bit more
+complex. When your ``DataFrame`` only has a single data type for all the
+columns, :atr:`DataFrame.values` will return the underlying data:
 
 .. ipython:: python
 
-    s.values
-    df.values
```

Comment:
reason you are not using ``.to_numpy()`` here? why advertise ``.values`` at all

Reply:
I think we should have it, and *maybe* put a note in the ``.values`` doc-string that its discourage (soft-deprecate it).

also we *could* do ``DataFrame.array`` -> ``[s.array for _, s in df.iteritems()]``
e.g. a list of the arrays

Reply:
I think a list of arrays wouldn't make sense, since that would violate the return type of `to_numpy()`.

I could alias `DataFrame.to_numpy` to just be DataFrame.values for now, and update doc references.